### ISSUE TITLE
`operator-assignment` should ignore `"use asm"` blocks

### ISSUE THREAD

**Issue by isiahmeadows** [eslint/eslint#8199]
Basically, when [`operator-assignment`](http://eslint.org/docs/rules/operator-assignment) is enabled, this is considered an error, when the "fix" is invalid asm.js:

```js
// Original
var f = (function () {
  "use asm";
  function f(x) {
    x = x | 0;
    return x | 0;
  }
  return f;
})();

// Fixed
var f = (function () {
  "use asm";
  function f(x) {
    x |= 0;
    return x | 0;
  }
  return f;
})();
```

The "fix" [syntactically invalidates the asm.js module](http://asmjs.org/spec/latest/#validatefunction-f), thus the rule should ignore it altogether, not marking it as a failure at all.

**Comment by not-an-aardvark** [comment]
The goal of the rule is to suggest replacing valid JavaScript with other valid JavaScript. If you need your code to run on engines with stricter semantics (e.g. asm.js) that don't allow compound assignments, I would recommend disabling this rule.

**Comment by isiahmeadows** [comment]
For the block, then?

**Comment by isiahmeadows** [comment]
Okay.
